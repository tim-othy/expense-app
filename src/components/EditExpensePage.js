import React from 'react';
import ExpenseForm from './ExpenseForm';

const EditExpensePage = () => (
  <div>
    <h1>Edit Expense</h1>
    <ExpenseForm
      onSubmit={(expense) => {
        console.log(expense)
      }}
    />
  </div>
);

export default EditExpensePage;