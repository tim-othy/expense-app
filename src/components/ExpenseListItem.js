export default (props) => (
  <div>
    {
      props.expenses.map(
        (expense) => (
          <p>{expense.description}</p>
          <p>{expense.amount}</p>
          <p>{expense.note}</p>
        )
      )
    }
  </div>
);
