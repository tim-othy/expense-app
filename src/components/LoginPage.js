import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


export const LoginPage = ({ startLogin }) => (
  <div className="centered-layout">
    <div className="centered-layout__box">
      <h1 className="centered-layout__title">Expense Tracker</h1>
      <button onClick={startLogin}>Login</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);