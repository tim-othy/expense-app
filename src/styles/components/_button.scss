.button {
    background: $blue;
    border: none;
    color: white;
    font-size: $l-font;
    font-weight: 300;
    padding: $s-size;
}