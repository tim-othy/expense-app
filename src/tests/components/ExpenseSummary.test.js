import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary correctly for 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={250} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly for multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={10} expensesTotal={250000} />);
  expect(wrapper).toMatchSnapshot();
  
});
