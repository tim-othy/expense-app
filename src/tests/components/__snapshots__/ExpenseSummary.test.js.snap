// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling  
    $2.50
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    10
     
    expenses
     totalling  
    $2,500.00
  </h1>
</div>
`;
